src/v/cluster/archival_metadata_stm.cc

```diff
@@ -267,6 +267,18 @@ ss::future<stm_snapshot> archival_metadata_stm::take_snapshot() {
 }
 
 model::offset archival_metadata_stm::max_collectible_offset() {
+    if (
+      _raft->log_config().has_overrides()
+      && !model::is_archival_enabled(
+        _raft->log_config().get_overrides().shadow_indexing_mode)) {
+        // The archival is disabled but the state machine still exists so we
```

Comment:
So the idea is to launch state machines for all partitions regardless of whether archival is enabled for them? I think that's ok... But we can then delete the TODO item in partition.cc :)